lib/backend.js

```diff
@@ -568,6 +573,86 @@ Backend.prototype.getChannels = function(collection, id) {
   ];
 };
 
+Backend.prototype.getSnapshot = function(agent, index, id, version, timestamp, callback) {
+  var backend = this;
+  this._getSnapshot(agent, index, id, version, timestamp, function (error, snapshot) {
+    if (error) return callback(error);
+
+    var request = {
+      collection: index,
+      id: id,
+      version: snapshot.version,
+      timestamp: snapshot.timestamp,
+      snapshots: snapshot.data ? [snapshot.data] : [],
+      type: snapshot.type
+    };
+
+    backend.trigger(backend.MIDDLEWARE_ACTIONS.readSnapshots, agent, request, function (error) {
+      if (error) return callback(error);
+      callback(null, {
+        data: request.snapshots[0],
+        version: request.version,
+        timestamp: request.timestamp,
+        type: request.type
+      });
+    });
+  });
+};
+
+Backend.prototype._getSnapshot = function (agent, index, id, version, timestamp, callback) {
+  version = (typeof version === 'number' && isFinite(version)) ? Math.max(0, version) : null;
+  var options = { metadata: true };
+  var timestampIsNumber = typeof timestamp === 'number';
```

Comment:
are we covered when we send NaN ?

Reply:
I think we would have been covered by the `if` block, because `1 > NaN === false`, but I've added an explicit check, too.